contracts/token/ERC20/extensions/ERC20uRWA.sol

```diff
@@ -0,0 +1,109 @@
+ */
+abstract contract ERC20uRWA is ERC20, ERC165, ERC20Freezable, ERC20Restricted, IERC7943 {
+    /// @inheritdoc ERC20Restricted
+    function isUserAllowed(address user) public view virtual override(IERC7943, ERC20Restricted) returns (bool) {
+        return super.isUserAllowed(user);
+    }
+
+    /// @inheritdoc ERC165
+    function supportsInterface(bytes4 interfaceId) public view virtual override(ERC165, IERC165) returns (bool) {
+        return interfaceId == type(IERC7943).interfaceId || super.supportsInterface(interfaceId);
+    }
+
+    /**
+     * @dev See {IERC7943-isTransferAllowed}.
+     *
+     * CAUTION: This function is only meant for external use. Overriding it will not apply the new checks to
+     * the internal {_update} function. Consider overriding {_update} accordingly to keep both functions in sync.
+     */
+    function isTransferAllowed(address from, address to, uint256, uint256 amount) external view virtual returns (bool) {
+        return (amount <= available(from) && isUserAllowed(from) && isUserAllowed(to));
+    }
+
+    /// @inheritdoc IERC7943
+    function getFrozen(address user, uint256) public view virtual returns (uint256 amount) {
+        return frozen(user);
+    }
+
+    /// @inheritdoc IERC7943
+    function setFrozen(address user, uint256, uint256 amount) public virtual {
+        _checkFreezer(user, amount);
+        require(amount <= balanceOf(user), ERC20InsufficientBalance(user, balanceOf(user), amount));
+        _setFrozen(user, amount);
+    }
+
+    /// @inheritdoc IERC7943
+    function forceTransfer(address from, address to, uint256, uint256 amount) public virtual {
+        _checkEnforcer(from, to, amount);
+        require(isUserAllowed(to), ERC7943NotAllowedUser(to));
+
+        // Update frozen balance if needed
+        uint256 currentFrozen = frozen(from);
+        uint256 currentBalance = balanceOf(from);
+        uint256 newAmount = currentBalance - amount;
+        if (currentFrozen > newAmount) {
+            _setFrozen(from, newAmount);
+        }
+
+        ERC20._update(from, to, amount); // Explicit raw update to bypass all restrictions
+        emit ForcedTransfer(from, to, 0, amount);
+    }
+
+    /**
+     * @dev See {ERC20-_update}.
+     *
+     * Requirements:
+     *
+     * * `from` and `to` must be allowed to transfer `amount` tokens (see {isTransferAllowed}).
+     */
+    function _update(
+        address from,
+        address to,
+        uint256 amount
+    ) internal virtual override(ERC20, ERC20Freezable, ERC20Restricted) {
+        if (from == address(0)) {
+            // Minting
+            require(isUserAllowed(to), ERC7943NotAllowedUser(to));
+        }
+        // Note that `isTransferAllowed` duplicates the `available` check made by `super._update` in ERC20Freezable,
+        // so, the following line is not needed but left for reference:
+        // require(isTransferAllowed(from, to, 0, amount), ERC7943NotAllowedTransfer(from, to, 0, amount));
```

Comment:
given that its just an external utility, I think its a bit misleading to bring up `isTransferAllowed` here